### PR TITLE
[BPK-3809] Show flare pointing upwards

### PR DESCRIPTION
**Pull request by georgegillams**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/master/Example/Backpack%20Screenshot/Screenshots.swift)
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/master/Backpack/Backpack.h)
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/FlareView/Classes/BPKFlarePath.h [hunk 1]
```diff
@@ -51,7 +53,9 @@ IB_DESIGNABLE @interface BPKFlarePath : NSObject
  *
  * @param size CGSize of the path required.
  */
-+ (UIBezierPath *)flareViewPathForSize:(CGSize)size NS_SWIFT_NAME(makeFlarePath(size:));
++ (UIBezierPath *)flareViewPathForSize:(CGSize)size
```
Comment (k0nserv) [inline Backpack/FlareView/Classes/BPKFlarePath.h:56]: Is this exposed externally?

File: Backpack/FlareView/Classes/BPKFlareView.m [hunk 2]
```diff
@@ -84,23 +84,53 @@ - (void)setup {
...
 
-    [self createLayerMask];
+- (void)setFlarePosition:(BPKFlarePosition)flarePosition {
+    if (_flarePosition != flarePosition) {
+        _flarePosition = flarePosition;
+        [self updateContentViewConstraints];
+        [self createLayerMask];
+
+        [self setNeedsLayout];
+        [self layoutIfNeeded];
```
Comment (k0nserv) [inline Backpack/FlareView/Classes/BPKFlareView.m:132]: This shouldn't be required `setNeedsLayout` already causes the view to be laid out in the next layout pass.

File: Example/SnapshotTests/BPKFlareViewSnapshotTest.m [hunk 3]
```diff
@@ -75,6 +75,18 @@ - (void)testFlareViewSnapshot {
     FBSnapshotVerifyView(parentView, nil);
 }
 
+- (void)testFlareViewSnapshotTop {
```
Comment (k0nserv) [inline Example/SnapshotTests/BPKFlareViewSnapshotTest.m:78]: Maybe we should rename `testFlareViewSnapshot`?

File: Backpack/FlareView/Classes/BPKFlareView.m (hunk 2 above)
Comment (georgegillams) [reply Backpack/FlareView/Classes/BPKFlareView.m:132]: This was needed to make the test work, but the example in the app seemed to be fine without it. Maybe I should move it to the test method only?

File: Backpack/FlareView/Classes/BPKFlarePath.h (hunk 1 above)
Comment (georgegillams) [reply Backpack/FlareView/Classes/BPKFlarePath.h:56]: It's in the `.h` file so I guess?

File: Example/SnapshotTests/BPKFlareViewSnapshotTest.m (hunk 3 above)
Comment (georgegillams) [reply Example/SnapshotTests/BPKFlareViewSnapshotTest.m:78]: That's what the default test is called though. Did you mean `testFlareViewTopSnapshot`?

File: Backpack/FlareView/Classes/BPKFlarePath.h (hunk 1 above)
Comment (k0nserv) [reply Backpack/FlareView/Classes/BPKFlarePath.h:56]: So then this is breaking?

File: Backpack/FlareView/Classes/BPKFlareView.m (hunk 2 above)
Comment (k0nserv) [reply Backpack/FlareView/Classes/BPKFlareView.m:132]: Yeah that sounds sensible

File: Example/SnapshotTests/BPKFlareViewSnapshotTest.m (hunk 3 above)
Comment (k0nserv) [reply Example/SnapshotTests/BPKFlareViewSnapshotTest.m:78]: Yeah I am saying we should rename that method given the introduction of `testFlareViewTopSnapshot`

Comment (georgegillams) [reply Example/SnapshotTests/BPKFlareViewSnapshotTest.m:78]: Oh I get it.

Clearly I need more coffee!

File: Backpack/FlareView/Classes/BPKFlarePath.h (hunk 1 above)
Comment (georgegillams) [reply Backpack/FlareView/Classes/BPKFlarePath.h:56]: Oh sorry my mistake - this is not imported into `FlareView.h`, so it's only publicly available if someone imports `Backpack/BPKFlarePath.h` which is unsupported

**Pull request merged**